src/core/init/create_media_source.ts

```diff
@@ -30,21 +30,35 @@ import {
 } from "../../compat";
 import { MediaError } from "../../errors";
 import log from "../../log";
+import Manifest from "../../manifest";
 import isNonEmptyString from "../../utils/is_non_empty_string";
 
 const { onSourceOpen$ } = events;
 
+/** Number of seconds in a regular year. */
+const YEAR_IN_SECONDS = 365 * 24 * 3600;
+
 /**
  * Set the media duration in the mediaSource.
  * @param {MediaSource} mediaSource
  * @param {number} duration
  */
 export function setDurationToMediaSource(
   mediaSource : MediaSource,
-  duration : number
+  manifest : Manifest
 ) : void {
-  const newDuration = duration === Infinity ? Number.MAX_VALUE :
-                                              duration;
+  const maximumPosition = manifest.getMaximumPosition();
+
+  // Some targets poorly support setting a very high number for durations.
+  // Yet, in live contents, we would prefer setting a value as high as possible
+  // to still be able to seek anywhere we want to (even ahead of the Manifest if
+  // we want to). As such, we put it at a safe default value of 2^32 excepted
+  // when the maximum position is already relatively close to that value, where
+  // we authorize exceptionally going over it.
+  const newDuration = !manifest.isLive ? maximumPosition :
+                                         Math.max(2 ^ 32,
```

Comment:
"^" is a XOR operator (2^32 = 34). You should use Math.pow to compute this.

Reply:
Ha, I'm dumb on that one. Did it too quickly. Should still work because of `Math.max` but yeah, not what was intended!

Thanks! And sorry.

Reply:
Fixed and squashed. Also fixed the ts error of having defined a `duration` property which was never used (in `load_on_media_source.ts`)